packages/core/src/state-management/anchor-resuming-service.ts

```diff
@@ -25,32 +25,33 @@ export class AnchorResumingService {
   }
 
   async resumeRunningStatesFromAnchorRequestStore(repository: Repository): Promise<void> {
-    this.#shouldBeClosed = false
+    if (this.#shouldBeClosed) {
+      throw Error('This AnchorResumingService is closed, create a new instance to resume')
+    }
+
     const currentTimestamp = Date.now()
     let gt: StreamID | undefined = undefined
     let batch = new Array<AnchorRequestStoreListResult>()
     do {
       batch = await repository.anchorRequestStore.list(RESUME_BATCH_SIZE, gt)
       batch.map((listResult) => {
         this.resumeQ.add(async () => {
-          const stream$ = await repository.load(listResult.key, {
-            sync: SyncOptions.PREFER_CACHE,
-          })
-          if ([AnchorStatus.FAILED, AnchorStatus.ANCHORED].includes(stream$.value.anchorStatus)) {
-            await repository.anchorRequestStore.remove(stream$.id)
-            return
-          }
-          // TODO: CDB-2010 Do we want to do the actual anchor request here? Or do we just want to start polling? If the latter, than .load(..) doesn't seem to start polling as of now.
-          await repository.stateManager.anchor(stream$)
-          if (listResult.value.timestamp > currentTimestamp) {
-            throw Error(
-              `System clock possibly moving backwards!!! Check your system clock and if it's accurate, delete your state store`
+          try {
+            await repository.fromMemoryOrStore(listResult.key)
```

Comment:
nice catch changing from `repository.load()` to `repository.fromMemoryOrStore()`.  The former might have generated a lot of unnecessary pubsub traffic